### ISSUE TITLE
Multiple blockers in process from anonymous booking (first order) to deletion of the order and deletion of account

### ISSUE THREAD

**Issue by mariobehling** [fossasia/open-event-frontend#6443]
1. When the user orders a ticket the first issue is, that the wrong email can shipw up for some reason mb@mariob... instead of mb@onte... (This could be happening for previously deleted accounts, or if a user was using the same browser in the past with another account?)
2. An unverified account with orders cannot be deleted. (Unexpected Error)
3. At the same time an order of an unverified account cannot be cancelled
4. And an order of a verified can neither be deleted
5. To be checked: Can an account be deleted if an event is in the past?
6. What should happen to the ticket data of the event? Can an organizer still see that? I think, yes. So, after a user cancels their tickets and deletes his or her account the order should show up still as "Canceled" and the user should simply be no longer in the system, but the order data of the canceled ticket stays with the organizer.

Basically even though there could be multiple issues that could be reported separately here, we need the entire process to be working from anonymous sign up to getting a ticket to verification and self-deletion of the account or to allowing unverified accounts to cancel their ticket and self-delete the account.

![Screenshot from 2021-01-20 12-57-20](https://user-images.githubusercontent.com/1583873/105175108-c9a5c780-5b23-11eb-98ea-66b0928c9fdf.png)

**Comment by iamareebjamal** [comment]
Orders cannot be deleted, only cancelled

**Comment by mariobehling** [comment]
In whatever way we define it, currently users can hardly ever achieve to delete their account. Even for past events it will block users. The entire process from getting a ticket to verifying and then participating in an event and finally deleting an account is not working as expected and we have a lot of users who want this. In order to have a service that people trust we need to enable people to delete their account.

This issue cannot be solve be just fixing some issue in the deletion process, but it needs to be tested from beginning to the end for different users, e.g. users with tickets, with events, without events, anonymous, verified etc. For example how can user who created an event delete their account? What should be the process here? Did we choose our policy in the right way?

Right now it seems whatever process exists users usually are not able to get their account deleted at all.

**Comment by iamareebjamal** [comment]
> When the user orders a ticket the first issue is, that the wrong email can shipw up for some reason mb@mariob... instead of mb@onte... (This could be happening for previously deleted accounts, or if a user was using the same browser in the past with another account?)

If you use autofill, it may have filled that email in the order form

> An unverified account with orders cannot be deleted.

The button shouldn't even be enabled for any account with orders, unverified or not

> At the same time an order of an unverified account cannot be cancelled

Any order created by non-admin/organizer account could not be cancelled, verified or not. Fixed here - https://github.com/fossasia/open-event-server/pull/7667

> To be checked: Can an account be deleted if an event is in the past?

If the user transfers the ownership, then yes

> What should happen to the ticket data of the event? Can an organizer still see that? I think, yes. So, after a user cancels their tickets and deletes his or her account the order should show up still as "Canceled" and the user should simply be no longer in the system, but the order data of the canceled ticket stays with the organizer.

Yes

**Comment by iamareebjamal** [comment]
> Basically even though there could be multiple issues that could be reported separately here, we need the entire process to be working from anonymous sign up to getting a ticket to verification and self-deletion of the account or to allowing unverified accounts to cancel their ticket and self-delete the account.

Tested

**Comment by iamareebjamal** [comment]
Reopening for verification

**Comment by iamareebjamal** [comment]
> how can user who created an event delete their account? What should be the process here?

Transfer the ownership to another user
